File: engine-taskserver-dotnet/src/main/dotnet/adapter/Job_impl.cs
namespace sos.spooler
{
    public class Job_impl
    {
        public Spooler spooler { get; private set; }
        public Job spooler_job { get; private set; }
        public Task spooler_task { get; private set; }
        public Log spooler_log { get; private set; }

        public Job_impl(Spooler contextSpooler, Job contextJob, Task contextTask, Log contextLog)
        {
            this.spooler = contextSpooler;
            this.spooler_job = contextJob;
            this.spooler_task = contextTask;
            this.spooler_log = contextLog;
        }

        public virtual bool spooler_init()
        {
            return true;
        }

        public virtual void spooler_exit()
        {
        }

        public virtual bool spooler_open()
        {
            return true;
        }

        public virtual void spooler_close()
        {
        }

        public virtual bool spooler_process()
        {
            return false;
        }

        public virtual void spooler_on_error()
        {
        }

        public virtual void spooler_on_success()
        {
        }
    }
}

File: sos/spooler/Job_impl.java
// $Id: Job_impl.java,v 1.2 2002/11/13 18:15:39 jz Exp $

package sos.spooler;

/**
 * Oberklasse für einen Spooler-Job.
 * Alle Methoden sind leer vorimplementiert.
 * Copyright (c) 2002, SOS GmbH Berlin
 * @author Joacim Zschimmer, Zschimmer GmbH
 * @version $Revision: 1.2 $
 */

public class Job_impl
{
    /** Wird bei use_engine="task" nur einmal für mehrere Jobläufe gerufen. 
      * Gegenstück ist {@link #spooler_exit()}.
      * @return false stoppt den Job.
      */
    public boolean  spooler_init        ()      { return true; }


    public void     spooler_exit        ()      {}
    public boolean  spooler_open        ()      { return true; }
    public void     spooler_close       ()      {}
    public boolean  spooler_process     ()      { return false; }
    public void     spooler_on_error    ()      {}
    public void     spooler_on_success  ()      {}


    public Log      spooler_log;
    public Task     spooler_task;
    public Job      spooler_job;
    public Thread   spooler_thread;
    public Spooler  spooler;
}
